napari/layers/utils/link_layers.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Iterable
+
+if TYPE_CHECKING:
+    from napari.layers import Layer
+
+#: Record of already linked layers... to avoid duplicating callbacks
+#  in the form of (id(layer1), id(layer2), attribute_name)
+_LINKED: set[tuple[int, int, str]] = set()
+
+
+# attributes that make no sense to link, or error on changing
+# we may want to include colormap here, but in most cases, it works fine.
+PROHIBITED = {'thumbnail', 'status', 'name'}
+
+
+def _get_common_evented_attributes(
+    layers: Iterable['Layer'],
+    prohibited: set[str] = PROHIBITED,
+    with_private=False,
+) -> set[str]:
+    """Get the set of common, non-private evented attributes in ``layers``.
+
+    Not all layer events are attributes, and not all attributes have
+    corresponding events.  Here we get the set of valid, non-private attributes
+    that are both events and attributes for the provided layer set.
+
+    Parameters
+    ----------
+    layers : iterable
+        A set of layers to evaluate for attribute linking.
+    prohibited : set, optional
+        A set of attributes to exclude, by default PROHIBITED
+    private : bool, optional
+        include private attributes
+
+    Returns
+    -------
+    names : set of str
+        A set of attribute names that may be linked between ``layers``.
+    """
+    from inspect import ismethod
+
+    try:
+        first_layer = next(iter(layers))
+    except StopIteration:
+        raise ValueError("``layers`` iterable must have at least one layer")
+    common_events = set.intersection(*(set(lay.events) for lay in layers))
+    common_attrs = set.intersection(*(set(dir(lay)) for lay in layers))
+    if not with_private:
+        common_attrs = {x for x in common_attrs if not x.startswith("_")}
+    common = common_events & common_attrs - prohibited
+
+    # lastly, discard any method-only events (we just want attrs)
+    for attr in set(common_attrs):
+        # properties do not count as methods and will not be excluded
+        if ismethod(getattr(first_layer.__class__, attr, None)):
+            common.discard(attr)
+
+    return common
+
+
+def experimental_link_layers(
```

Comment:
Can you also provide an `experimental_unlink_layers`. One approach might be to let it take either a single layers iterable and unlink all the connection it can find between those layers or if two layers iterables are given then unlink all the connections between the first list and the second list. That could be quite nice if you say want to disconnect one layer from a group of others without disrupting connections within that group.

Reply:
definitely, was just working on it! :)  will also add a context manager.

Reply:
ok, added `unlink_layers` and a `linked_layers` context manager.  i didn't do two lists of iterables: i did a single list.  If you provide a single layer, it will be unlinked from all other layers, if you provide a list, all combinations will be unlinked,  is that going to meet your imagined use case here?

Reply:
yeah, i think that will work, i'm imagining the clicking behavior in the layer context menus and an option like `unlink layers` if you have one selected and `unlink layers` if you have multiple selected and I think that's the behavior i'd want to see

Reply:
yeah, I think this should work then.  basically, if you click on the "lock" icon on any individual layer, it will call `unlink_layer([layer])` and "free" it from connections